.github/workflows/check-binary-size.yml

```diff
@@ -6,9 +6,12 @@ name: Check binary size
 
 on:
   pull_request_target:
-    # HACK(jubilee): something broke the distributed LLVM libso and I don't know what.
-    branches: []
-#      - master
+    branches:
+      - master
+
+# Remove all permissions for the workflow to increase security; the individual jobs
+# below specify the exact permissions they need
+permissions: {}
```

Comment:
If the jobs specify permissions then do they not strip permissions from the jobs? Because specifying any one permission removes all others.

In either case it seems somewhat irrelevant. It's a false safety if a permission later in the job can be readded, no? Please be more precise in what we are achieving here. What is the permission we are preventing from being misused?

Reply:
I have extended the PR description to explain this. This requires that every job explicitly needs to list additional permissions it needs, and prevents that in the future by accident a job without `permissions` is added which then gets too extensive default permissions.